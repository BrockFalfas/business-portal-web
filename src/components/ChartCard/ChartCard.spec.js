import React from 'react';
import { shallow } from 'enzyme';

import { ChartCard } from './index';

const fakeThemeColors = {
  accent: '#ffffff',
  chartColor: '#ff0000',
  text: '#000000',
};

const initChartCard = overrides => {
  return shallow(<ChartCard themeColors={fakeThemeColors} {...overrides} />);
};

describe('component: <ChartCard/>', () => {
  it('should render without crashing', () => {
    const wrapper = initChartCard();
    expect(wrapper).toBeTruthy();
  });

  it('should use theme colors', () => {
    const wrapper = initChartCard();
    expect(wrapper.find('.ChartCard-title').prop('style').color).toBe(fakeThemeColors.text);
    expect(wrapper.find('.ChartCard-aggregated-value').prop('style').color).toBe(
      fakeThemeColors.accent
    );
    expect(wrapper.find('Icon').prop('style').color).toBe(fakeThemeColors.accent);
  });

  it('should render title and aggregated value', () => {
    const wrapper = initChartCard({
      aggregatedValue: '$123,456.78',
      title: 'Revenue',
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should pass themeColors and all other props to the chart', () => {
    const chartProps = {
      other1: 'value1',
      other2: 'value2',
    };
    const wrapper = initChartCard(chartProps);
    const chart = wrapper.find('.ChartCard-chart').childAt(0);
    expect(chart.props()).toMatchObject({
      ...chartProps,
      themeColors: fakeThemeColors,
    });
  });
});
