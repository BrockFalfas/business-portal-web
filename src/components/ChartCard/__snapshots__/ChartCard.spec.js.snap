// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: <ChartCard/> should render title and aggregated value 1`] = `
<div
  className="ChartCard"
>
  <header
    className="ChartCard-header"
  >
    <div
      className="ChartCard-title"
      style={
        Object {
          "color": "#000000",
        }
      }
    >
      <Icon
        style={
          Object {
            "color": "#ffffff",
          }
        }
        type="area-chart"
      />
       
      Revenue
    </div>
    <div
      className="ChartCard-aggregated-value"
      style={
        Object {
          "color": "#ffffff",
        }
      }
    >
      $123,456.78
    </div>
  </header>
  <div
    className="ChartCard-chart"
  >
    <withResponsiveness(withTheme(ChartComponent))
      themeColors={
        Object {
          "accent": "#ffffff",
          "chartColor": "#ff0000",
          "text": "#000000",
        }
      }
    />
  </div>
</div>
`;
